docs/troubleshooting/README.md

```diff
@@ -1 +1,611 @@
+      reason: ApplySucceeded
+      status: "True"
+      type: ResourceApplied
+  - conditions:
+    - lastTransitionTime: "2023-11-27T20:25:40Z"
+      message: 'kind-cluster-2 is not selected: ClusterUnschedulable, none of the
+        nonempty required cluster affinity term (total number: 1) is matched'
+      observedGeneration: 2
+      reason: ScheduleFailed
+      status: "False"
+      type: ResourceScheduled
+  selectedResources:
+  ...
+```
+
+**ClusterResourcePlacementScheduled** is set to **false** because we want to pick two clusters with label **env:prod** but only one member cluster has the correct label mentioned in **clusterAffinity**
+
+We can also take a look at the **ClusterSchedulingPolicySnapshot** status to figure out why the scheduler could not schedule the resource for the placement policy specified.
+
+The corresponding **ClusterSchedulingPolicySnapshot's** spec and status gives us even more information why scheduling failed,
+
+```
+apiVersion: placement.kubernetes-fleet.io/v1beta1
+kind: ClusterSchedulingPolicySnapshot
+metadata:
+  annotations:
+    kubernetes-fleet.io/CRP-generation: "2"
+    kubernetes-fleet.io/number-of-clusters: "2"
+  creationTimestamp: "2023-11-27T21:33:01Z"
+  generation: 1
+  labels:
+    kubernetes-fleet.io/is-latest-snapshot: "true"
+    kubernetes-fleet.io/parent-CRP: crp-4
+    kubernetes-fleet.io/policy-index: "0"
+  name: ...
+  ownerReferences:
+  - apiVersion: placement.kubernetes-fleet.io/v1beta1
+    blockOwnerDeletion: true
+    controller: true
+    kind: ClusterResourcePlacement
+    name: ...
+    uid: 37e83327-26e0-4c48-8276-e62cc6aa067f
+  resourceVersion: "10085"
+  uid: f2a3d0ea-c9fa-455d-be09-51b5d090e5d6
+spec:
+  policy:
+    affinity:
+      clusterAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          clusterSelectorTerms:
+          - labelSelector:
+              matchLabels:
+                env: prod
+    placementType: PickN
+  policyHash: ZjE0Yjk4YjYyMTVjY2U3NzQ1MTZkNWRhZjRiNjQ1NzQ4NjllNTUyMzZkODBkYzkyYmRkMGU3OTI3MWEwOTkyNQ==
+status:
+  conditions:
+  - lastTransitionTime: "2023-11-27T21:33:01Z"
+    message: could not find all the clusters needed as specified by the scheduling
+      policy
+    observedGeneration: 1
+    reason: SchedulingPolicyUnfulfilled
+    status: "False"
+    type: Scheduled
+  observedCRPGeneration: 2
+  targetClusters:
+  - clusterName: kind-cluster-1
+    clusterScore:
+      affinityScore: 0
+      priorityScore: 0
+    reason: picked by scheduling policy
+    selected: true
+  - clusterName: kind-cluster-2
+    reason: 'ClusterUnschedulable, none of the nonempty required cluster affinity
+      term (total number: 1) is matched'
+    selected: false
+```
+
+The solution here is to add the **env:prod** label to the member cluster resource for **kind-cluster-2** as well so that the scheduler can pick the cluster to propagate resources.
+
+### How to verify the latest ClusterSchedulingPolicySnapshot for a CRP?
```

Comment:
Could you please move all of the validating/checking questions to the bottom? they're more like debugging tools/apis.

How to verify the latest ClusterSchedulingPolicySnapshot for a CRP?
How to find the latest ClusterResourceBinding resource?
...